lib/purescript-cst/src/Language/PureScript/Types.hs

```diff
@@ -43,6 +43,17 @@ instance NFData SkolemScope
 instance Serialise SkolemScope
 
 -- |
+-- Metadata for determining whether or not a type variable is used for
+-- visible type applications.
+--
+data VtaTypeVar
+  = IsVtaTypeVar
+  | NotVtaTypeVar
```

Comment:
What do you think about adding another variant here `IsVtaTypeVarRequired`, which can be derived from the result of an ambiguity check. Then whenever instantiating a required vta implicitly, we can throw an appropriate error?